# origin: kaos (BitKeeper)
# cset: 1.1482.2.3 (2.4) key=4189fe8bQyYTlpITPgFN0mT6orB-Pw
# inclusion: upstream
# description: [PATCH] Avoid oops in proc_delete_inode
# revision date: Fri, 26 Nov 2004 14:59:39 +0900
#
# S rset: ChangeSet|1.1482.2.2..1.1482.2.3
# I rset: fs/proc/base.c|1.19..1.20
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/11/04 08:03:55-02:00 kaos@sgi.com 
#   [PATCH] Avoid oops in proc_delete_inode
#   
#   Under heavy load, vmstat, top and other programs that access /proc can
#   oops.  PROC_INODE_PROPER(inode) is sometimes false for pid entries
#   (usually zombies), but inode->u.generic_ip is not NULL.
#   
#   Backport a fix by AL Viro from 2.5.7-pre2 to 2.4.28-rc1.
#   
#   Signed-off-by: Keith Owens <kaos@sgi.com>
#   
#   Index: 2.4.28-rc1/fs/proc/base.c
#   ===================================================================
# 
# fs/proc/base.c
#   2004/11/04 00:25:16-02:00 kaos@sgi.com +1 -0
#   Avoid oops in proc_delete_inode
# 
#
===== fs/proc/base.c 1.19 vs 1.20 =====
--- 1.19/fs/proc/base.c	2004-07-30 22:29:39 +09:00
+++ 1.20/fs/proc/base.c	2004-11-04 11:25:16 +09:00
@@ -780,6 +780,7 @@
 	return inode;
 
 out_unlock:
+	node->u.generic_ip = NULL;
 	iput(inode);
 	return NULL;
 }
